### DISCUSSION TITLE
Delegation

### DISCUSSION THREAD

**Discussion by prologic** [balzack/databag#53] [Ideas]
We did this in [Salty IM](https://salty.im) (_very similar goals to your project here, but you have a shiny mobile app 🤣_)

The basic idea is to use SRV records to delegate the "databag" service to a different host.domain[:port] as required. That way I can run databag on say databag.mydomain.tld but have an identity of prologic@mydomain.tld

**Comment by balzack** [comment]
Your project looks really cool! I'm trying to do the quick start but getting this error:
/usr/local/go/src/runtime/internal/sys/consts.go:13:7: StackGuardMultiplier redeclared in this block

It's probably some noob mistake. I was running go 1.17 and had to upgrade to 1.19, perhaps I did something wrong.

**Reply by prologic** [reply]
Can't say that I've seen that error before 🤔 Out `go.mod` definately calls for _at least_ Go 1.19:

```#!console
prologic@Jamess-Mac-Studio
Sun Jul 23 20:44:58
~/Projects/saltyim/saltyim
 (master) 130
$ head go.mod
module go.salty.im/saltyim

go 1.19
```

So I'd start there, maybe try to build from source if you're curious. In any case the part I'm talking about in terms of delegration is in our [spec](https://salty.im/spec.html) (_search for "Delegation")
